Tutorials/whatiswot/6-JSON_Schema_in_Practice/6-JSON-Schema-in-Practice.vtt

```diff
@@ -0,0 +1,131 @@
+WEBVTT
+
+00:00:00.000 --> 00:00:06.793 
+As we discussed in the previous video, JSON Schema is a declarative format for “describing
+
+00:00:06.793 --> 00:00:13.587 
+the structure of other data". Now we will show how it looks like in practice.
+
+00:00:13.587 --> 00:00:20.215 
+There are many different drafts of JSON Schema, and it is not always easy to
+
+00:00:20.215 --> 00:00:26.843 
+tell which draft it is. Thus, we use the schema keyword to declare which version
+
+00:00:26.843 --> 00:00:33.470 
+is used. WoT uses Draft7, that's why throughout this tutorial we will use Draft7.
+
+00:00:33.470 --> 00:00:42.970 
+In JSON Schema, an empty object is a completely valid schema that will accept any valid JSON.
+
+00:00:42.970 --> 00:00:51.395 
+It can be also used "true" in a place of the empty object to represent a schema that matches anything,
+
+00:00:51.395 --> 00:00:59.883 
+or "false" for a schema that matches nothing.
+
+00:00:59.883 --> 00:01:03.033 
+Now let's look at the Schema validation keywords.
+
+00:01:03.033 --> 00:01:07.514 
+One of the most useful common things to do in a JSON
+
+00:01:07.514 --> 00:01:11.995 
+Schema is to restrict a specific type by using the "type" keyword.
+
+00:01:11.995 --> 00:01:18.483 
+There are two numeric types in JSON Schema: "integer" and "number". They
+
+00:01:18.483 --> 00:01:24.970 
+share the same validation keywords. "Number" accepts integers and floating numbers.
+
+00:01:24.970 --> 00:01:30.333 
+Whereas "integer" rejects floating points.
+
+00:01:30.333 --> 00:01:41.045 
+Objects are the mapping type in JSON. They map “keys” to “values”. In JSON, the “keys” must always be strings.
+
+00:01:41.045 --> 00:01:48.825 
+The properties which are a key-value pair on an object are defined using the properties keyword. The value of "properties"
+
+00:01:48.825 --> 00:01:56.604 
+is an object, where each key is the name of a property and each value is a schema used to
+
+00:01:56.604 --> 00:02:04.383 
+validate that property. The second one is not accepted because the value of the number property is a string.
+
+00:02:04.383 --> 00:02:12.087 
+In addition to the last example, The "additionalProperties" keyword can be used to control the handling of extra
+
+00:02:12.087 --> 00:02:19.791 
+keys, which is, properties whose names are not listed in the "properties" keyword. If it is false, then
+
+00:02:19.791 --> 00:02:27.495 
+it does not accept additional properties. Otherwise, you can specify what type can be given too.
+
+00:02:27.495 --> 00:02:33.552 
+By default, the "properties" defined by the properties keyword are not required. However,
+
+00:02:33.552 --> 00:02:39.608 
+one can provide a list of required properties using the "required" keyword.
+
+00:02:39.608 --> 00:02:45.695 
+Since the second one does not contain email information it is not accepted.
+
+00:02:45.695 --> 00:02:56.620 
+Arrays are used for ordered elements. In JSON, each element in an array can be of a different type.
```

Comment:
The timing of the animation to show what is valid and what is not is a little bit off. The video stays silent with not animation then shortly shows the X and ✔️ sing. It is much quicker than the other animations. I know that it is a pain to edit now so I'll leave that to you to decide where to move on or fix it.

Reply:
It is perfect now, thanks!